Preprocessing/AudioNormalizer.py

```diff
@@ -0,0 +1,104 @@
+import argparse
+from pathlib import Path
+import sys
+from typing import (
+    List,
+    Optional,
+)
+
+import ffmpeg
+
+from Preprocessing.ErrorHandlingLogger import ErrorHandlingLogger
+from Preprocessing.utils import setup_logger
+
+
+class AudioNormalizer:
+    SUPPORTED_EXTENSIONS: List[str] = [".mp4", ".mkv", ".avi"]
+
+    def __init__(self, input_folder: str, output_folder: str):
+        self.input_folder = Path(input_folder)
+        self.output_folder = Path(output_folder)
+        self.logger = ErrorHandlingLogger(
+            class_name=self.__class__.__name__,
+            logger=setup_logger(self.__class__.__name__),
+        )
+
+    def run(self) -> int:
+        self.logger.info("Starting audio normalization...")
+        try:
+            self.process_folder()
+        except Exception as e:
+            self.logger.error(f"Unexpected critical error in run: {e}")
+        return self.logger.finalize()
+
+    def process_folder(self) -> None:
+        try:
+            if not self.input_folder.is_dir():
+                self.logger.error(f"Invalid input folder: {self.input_folder}")
+                raise ValueError(f"Invalid input folder: {self.input_folder}")
+
+            for video_file in self.input_folder.rglob("*"):
+                if video_file.suffix.lower() in self.SUPPORTED_EXTENSIONS:
+                    self.process_video_file(video_file)
+        except Exception as e:
+            self.logger.error(f"Unexpected error in process_folder: {e}")
```

Comment:
A nie no to tutaj chyba nie musisz tylu try-catchów trzaskać, mi chodziło tylko o run i ewentualnie dla konkretnego pliku, żeby mieć jakiś kontekst błędu, a nie dosłownie w każdej funkcji